nautobot/core/models/__init__.py

```diff
@@ -50,3 +97,92 @@ def validated_save(self):
         """
         self.full_clean()
         self.save()
+
+    def natural_key(self):
+        """
+        Smarter default implementation of natural key construction.
+
+        1. Handles nullable foreign keys (https://github.com/wq/django-natural-keys/issues/18)
+        2. Handles variadic natural-keys (e.g. Location model - [name, parent__name, parent__parent__name, ...].)
+        """
+        vals = []
+        for name_list in [name.split("__") for name in self.get_natural_key_fields()]:
+            val = self
+            for attr_name in name_list:
+                val = getattr(val, attr_name)
+                if val is None:
+                    break
+            vals.append(val)
+        # Strip trailing Nones from vals
+        cleaned_vals = []
+        for val in reversed(vals):
+            if val is None and not cleaned_vals:
+                continue
+            cleaned_vals = [val] + cleaned_vals
+        return cleaned_vals
+
+    natural_key_separator = "/"
+
+    @property
+    def natural_key_slug(self):
+        """Less naive implementation than django-natural-keys provides by default, based around URL percent-encoding."""
```

Comment:
Add a note that this also renders `None` values as encoded null characters (`%00`).